### PR TITLE
Add documentation on slack mentions

### PR DESCRIPTION
**Pull request by phongulus**
Add some documentation on setting up Slack mentions and expected behaviour.

### CODE REVIEW

File: documentation/config_docs.md [hunk 1]
```diff
@@ -41,6 +45,9 @@ Refer [here](https://docs.github.com/en/free-pro-team@latest/developers/webhooks
 | `status_rules` | status rules config object | all status notifications are ignored |
 | `project_owners` | project owners config object | no project owners are defined |
 | `ignored_users` | list of users to be ignored on all notifications | no user is ignored |
+| `user_mappings` | list of mappings from git email and/or GitHub handle to Slack email | no mapping defined
+
+Note that in `user_mappings`, git email to Slack email mappings are used for status DMs, while GitHub handle to Slack email mappings are used to get Slack mentions in notifications.
```
Comment (rr0gi) [inline documentation/config_docs.md:50]: not explained why need to map by handle when email also available, the latter should be always better?

**Review by rr0gi — CHANGES_REQUESTED** [review]

File: documentation/config_docs.md (hunk 1 above)
Comment (phongulus) [reply documentation/config_docs.md:50]: Email is only available for commits and pushes, and it's the one from `git config user.email` if I understand correctly, and that's what I meant by "git email" in the readme. So it's possible to match between emails for CI status DMs. In other types of comments we only have access to the GitHub handle. I can make this clearer in documentation.

Comment (tysg) [reply documentation/config_docs.md]: I think @rr0gi meant to explain the design considerations and decisions on the "GitHub handle to Slack email mappings are used to get Slack mentions in notifications" part, i.e. why not use git email to Slack email too?

**Comment by Khady** [comment]
@phongulus anything missing to merge this PR?

**Comment by phongulus** [comment]
Sorry, I completely forgot about this. I'll look again, add architectural comments as per @tysg's suggestion and re-request review.

**Comment by phongulus** [comment]
Sorry for the delay. @Khady for some reason I can't request a review from you. Can you take a look, and if it looks OK i think we can merge.

**Review by rr0gi — APPROVED** [review]

**Pull request merged**